### DISCUSSION TITLE
Requirements restructuring between Base Site and Recipe Base Site

### DISCUSSION THREAD

**Discussion by muskie9** [dynamic/silverstripe-base-site#126] [Ideas]
Along with the next major releases of base site and recipe base site, we'll be doing an exercise to ensure package requirements are organized for best use cases based on whether the recipe is used or not in a website. Below are the requirements per module/recipe:

**dynamic/silverstripe-base-site**
Note: `innoweb/silverstripe-social-meta` is a new addition 

```json
    "require": {
        "axllent/silverstripe-email-obfuscator": "^2.0",
        "axllent/silverstripe-scaled-uploads": "^2.1",
        "dnadesign/silverstripe-elemental": "^4.0",
        "dynamic/silverstripe-geocoder": "^1.1",
        "dynamic/silverstripe-site-tools": "^1.0",
        "innoweb/silverstripe-social-metadata": "^7.0",
        "jonom/silverstripe-betternavigator": "^4.0",
        "jonom/silverstripe-text-target-length": "^2.0",
        "silverstripe/blog": "^3.0",
        "silverstripe/recipe-cms": "^4.3",
        "silverstripe/sharedraftcontent": "^2.0",
        "silverstripe/userforms": "^5.0",
        "tractorcow/silverstripe-sitemap2": "^4.0",
        "unclecheese/display-logic": "^2.0",
        "wilr/silverstripe-googlesitemaps": "^2.1"
    }
```

**dynamic/recipe-silverstripe-base-site**
```json
    "require": {
        "php": ">=7.0.0",
        "axllent/silverstripe-cms-tweaks": "^2.0",
        "axllent/silverstripe-bootstrap-forms": "^2.0",
        "axllent/silverstripe-version-truncator": "^2.0",
        "dynamic/silverstripe-base-site": "^4.0",
        "dynamic/recipe-silverstripe-elemental-base-site": "^1.0",
        "fractas/elemental-stylings": "^1.0@RC",
        "silverstripe/elemental-bannerblock": "^2.0",
        "silverstripe/recaptcha": "^3.0",
        "silverstripe/redirectedurls": "^2.0",
        "silverstripe/securityreport": "^2.0",
        "silverstripe/spamprotection": "^3.0",
        "silverstripe/widgets": "^2.0",
        "tractorcow/silverstripe-robots": "^4.0"
    }
```

- What modules would be better suited in base site vs the recipe?
- What modules would be better suited as suggestions?
- What modules require configuration?
- What modules will be deprecated as we use new modules (for example SiteTools/CompanyConfig)?

The above are some examples of thoughts that will drive where packages live and what migration tools will be needed to move data to newer modules now used.

**Comment by jsirish** [comment]
if we want to keep Base Site super clean, we could move any modules with page types to recipe:

* blog
* userforms
* sitemap2

I believe the following could be removed from base site:
* geocoder

I believe the following could be removed from recipe base site:
* axllent/silverstripe-cms-tweaks
* silverstripe/recaptcha
* silverstripe/widgets (could be considered per-project)

**Comment by muskie9** [comment]
**Proposed Base Site Requirements**'

```json
{
    "axllent/silverstripe-email-obfuscator": "^2.0",
    "axllent/silverstripe-scaled-uploads": "^2.1",
    "dnadesign/silverstripe-elemental": "^4.0",
    "innoweb/silverstripe-social-metadata": "^7.0",
    "jonom/silverstripe-betternavigator": "^4.0",
    "jonom/silverstripe-text-target-length": "^2.0",
    "silverstripe/sharedraftcontent": "^2.0",
    "silverstripe/spamprotection": "^3.0",
    "unclecheese/display-logic": "^2.0",
    "wilr/silverstripe-googlesitemaps": "^2.1",
    "axllent/silverstripe-bootstrap-forms": "^2.0",
    "axllent/silverstripe-version-truncator": "^2.0",
    "silverstripe/redirectedurls": "^2.0",
    "silverstripe/spamprotection": "^3.0",
    "tractorcow/silverstripe-robots": "^4.0"
}
```

**Reply by muskie9** [reply]
Spam protection likely can go away/stay in recipe as userforms (what we primarily use it for) would not be in base site in this case.
